### ISSUE TITLE
Bun install crash (Windows, can't reproduce)

### ISSUE THREAD

**Issue by Razunter** [oven-sh/bun#12118]
### How can we reproduce the crash?

Can't reproduce, worked on the second try

### Relevant log output

```shell
bun add v1.1.16 (bf7b327f)
============================================================
Bun v1.1.16 (bf7b327f) Windows x64
Args: "bun" "install" "-D" "postcss-load-config"
Features: dotenv(2) extracted_packages(4)
Elapsed: 869ms | User: 15ms | Sys: 78ms
RSS: 121.80MB | Peak: 121.80MB | Commit: 0.20GB | Faults: 30111

panic(main thread): index out of bounds: index 3196, len 3093
```

### Stack Trace (bun.report)

Bun v1.1.16 ([`bf7b327`](<https://github.com/oven-sh/bun/tree/bf7b327f68568104a5e40c1944b7b255b02f0745>)) on windows x86_64 [InstallCommand]

**panic**: index out of bounds: index 3196, len 3093

- [`install.zig:4906`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/install/install.zig#L4906>): `enqueueDependencyWithMain`
- [`install.zig:5793`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/install/install.zig#L5793>): `processDependencyList`
- [`install.zig:12861`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/install/install.zig#L12861>): `isDone`
- [`install.zig:12899`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/install/install.zig#L12899>): `installPackages`
- [`install.zig:13637`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/install/install.zig#L13637>): `installWithManager`
- [`install.zig:10031`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/install/install.zig#L10031>): `updatePackageJSONAndInstallWithManagerWithUpdates`
- [`filter_run.zig:413`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/cli/filter_run.zig#L413>): `install`
- [`cli.zig:1494`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/cli.zig#L1494>): `start`
- [`cli.zig:62`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/cli.zig#L62>): `start`
- [`main.zig:51`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/main.zig#L51>): `main`

**Comment by dylan-conway** [comment]
Can you share the `package.json` and `bun.lockb` for this project?

**Comment by Razunter** [comment]
[package.json](https://github.com/user-attachments/files/15960640/package.json)
[bun.lockb.zip](https://github.com/user-attachments/files/15960645/bun.lockb.zip)

**Comment by dylan-conway** [comment]
Having trouble reproducing this bug with your package.json and lockfile. Do you have this crash with each install?

**Comment by dylan-conway** [comment]
Fixed in bun v1.1.19 with #12485
